operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakRealmJob.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.operator.v2alpha1;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.batch.v1.Job;
+import io.fabric8.kubernetes.api.model.batch.v1.JobBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.utils.KubernetesResourceUtil;
+import io.quarkus.logging.Log;
+import org.keycloak.operator.OperatorManagedResource;
+import org.keycloak.operator.v2alpha1.crds.KeycloakRealm;
+import org.keycloak.operator.v2alpha1.crds.KeycloakRealmStatusBuilder;
+
+import java.util.List;
+import java.util.Optional;
+
+public class KeycloakRealmJob extends OperatorManagedResource {
```

Comment:
We should re-use here the `KeycloakDeployment` as we're currently duplicating some of its logic here. E.g. the RealmJob probably should not try to know the name of the main KeycloakDeployment nor how to restart it. But for the prototype this is fine and we can refactor as a follow-up.

Reply:
This was my initial intention, unfortunately the `KeycloakDeployment` at the moment takes quite some arguments to be constructed which are not needed at all by this controller ...
I agree we can and should refactor later on.

Reply:
👍  Please mark this issue as resolved.